stake-pool/single-pool/src/processor.rs

```diff
@@ -856,23 +983,17 @@ impl Processor {
             return Err(SinglePoolError::SignatureMissing.into());
         }
 
-        // checking the mint exists confirms pool is initialized
-        {
-            let pool_mint_data = pool_mint_info.try_borrow_data()?;
-            let _ = Mint::unpack_from_slice(&pool_mint_data)?;
-        }
-
-        let vote_address_str = vote_account_address.to_string();
-        let token_name = format!("SPL Single Pool {}", &vote_address_str[0..15]);
-        let token_symbol = format!("st{}", &vote_address_str[0..7]);
+        let pool_address_str = pool_info.key.to_string();
+        let token_name = format!("SPL Single Pool {}", &pool_address_str[0..15]);
+        let token_symbol = format!("st{}", &pool_address_str[0..7]);
```

Comment:
the idea i had for this was:
* it feels more "appropriate" that the pool identity is the pool state account
* this lets me do a flow in the cli where the user fetches their token accounts, and fetches the single pool program accounts, and compares the addresses to determine which correspond to which, without needing to parse vote accounts

Reply:
Sorry, I'm not understanding the second part.  My comment has to do with the default pool token metadata using the pool address rather than the validator address. I guess the validator could always update it to whatever they want, but for someone seeing their tokens in a wallet, it could be nice to have the validator address in it by default.

Definitely agreed that you need the state accounts, they'll be a huge help to clients.

Reply:
actually nevermind, i was overthinking something very simply done a different way, and it doesnt matter because i forgot about the whole "the metadata can be updated to something else" part